python-package/lightgbm/engine.py

```diff
@@ -131,12 +131,16 @@ def train(params, train_data, num_boost_round=100,
         predictor = init_model
     else:
         predictor = None
+    init_iteration = predictor.init_iteration if predictor else 0
```

Comment:
I fixed it. Seem init_iteration is not needed in Booster, I already test on init_model=Booster, model_file and Predictor.